src/app/base/components/bricks-integrator/bricks-integrator.component.ts

```diff
@@ -340,5 +344,13 @@ export class BricksIntegratorComponent implements OnInit, OnDestroy {
     if (this.newTaskId.observers.length > 0) this.newTaskId.emit(taskId);
   }
 
-
+  checkIfFunctionNameIsUnique(name: string) {
+    this.codeParser.nameTaken = this.nameLookups?.find(x => x == name) != undefined;
+    name = toPythonFunctionName(name);
+    this.checkCanAccept();
+    if (this.config.preparedCode) {
+      this.codeParser.functionName = name;
+      this.config.preparedCode = this.config.preparedCode.replace(getPythonFunctionName(this.config.preparedCode), name);
```

Comment:
shouldn't work as expected because of replace issues. Also, the canAccept variable only seems to be set in replaceVariables function which isn't called here (which is correct). But the accept state isn't checked so we should create a function to check the state and call it

Reply:
ok didn't read fully. The checkCanAccept function is the correct one. Also it should be run at the end imo.